File: test/python/test_json_primitives.py
"""Unit tests for the JSON key primitives (rename/erase) used by schema migrations."""

from stochtree.serialization import JSONSerializer


def test_rename_and_erase_top_level():
    s = JSONSerializer()
    s.add_scalar("old", 3.5)

    s.rename_field("old", "new")
    assert not s.contains("old")
    assert s.contains("new")
    assert s.get_scalar("new") == 3.5

    s.erase_field("new")
    assert not s.contains("new")


def test_rename_and_erase_subfolder():
    s = JSONSerializer()
    s.add_scalar("old", 7.0, subfolder_name="sub")

    s.rename_field("old", "new", subfolder_name="sub")
    assert not s.contains("old", subfolder_name="sub")
    assert s.contains("new", subfolder_name="sub")
    assert s.get_scalar("new", subfolder_name="sub") == 7.0

    s.erase_field("new", subfolder_name="sub")
    assert not s.contains("new", subfolder_name="sub")


def test_rename_and_erase_missing_is_noop():
    s = JSONSerializer()
    # Neither should raise on an absent field.
    s.rename_field("nope", "whatever")
    s.erase_field("nope")
    assert not s.contains("whatever")
    assert not s.contains("nope")

File: test/R/testthat/test-json-primitives.R
# Unit tests for the JSON key primitives (rename/erase) used by schema migrations.

test_that("CppJson rename_field / erase_field (top-level)", {
  skip_on_cran()
  j <- createCppJson()
  j$add_scalar("old", 3.5)

  j$rename_field("old", "new")
  expect_false(j$contains("old"))
  expect_true(j$contains("new"))
  expect_equal(j$get_scalar("new"), 3.5)

  j$erase_field("new")
  expect_false(j$contains("new"))
})

test_that("CppJson rename_field / erase_field (subfolder)", {
  skip_on_cran()
  j <- createCppJson()
  j$add_scalar("old", 7.0, subfolder_name = "sub")

  j$rename_field("old", "new", subfolder_name = "sub")
  expect_false(j$contains("old", subfolder_name = "sub"))
  expect_true(j$contains("new", subfolder_name = "sub"))
  expect_equal(j$get_scalar("new", subfolder_name = "sub"), 7.0)

  j$erase_field("new", subfolder_name = "sub")
  expect_false(j$contains("new", subfolder_name = "sub"))
})

test_that("CppJson rename/erase of a missing field is a no-op", {
  skip_on_cran()
  j <- createCppJson()
  expect_silent(j$rename_field("nope", "whatever"))
  expect_silent(j$erase_field("nope"))
  expect_false(j$contains("whatever"))
})
